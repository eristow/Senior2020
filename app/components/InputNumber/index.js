/**
 *
 * InputNumber
 *
 */

import styled from 'styled-components';
import PropTypes from 'prop-types';

const InputNumber = styled.input.attrs(props => ({
  type: 'number',
  padding: props.padding || '0.5em',
  width: props.width || '5em',
}))`
  display: block;
  color: black;
  border: 2px solid deepskyblue;
  border-radius: 3px;
  font-size: 1em;
  padding: ${props => props.padding};
  width: ${props => props.width};
  margin: 0.5em 0.5em;
`;

InputNumber.propTypes = {
  padding: PropTypes.string,
  width: PropTypes.string,
};

export default InputNumber;
