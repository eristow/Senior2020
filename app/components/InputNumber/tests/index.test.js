/**
 *
 * Tests for InputNumber
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import InputNumber from '../index';

describe('<InputNumber />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<InputNumber />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Expect to have additional unit tests specified', () => {
    expect(true).toEqual(false);
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it.skip('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<InputNumber />);
    expect(firstChild).toMatchSnapshot();
  });
});
