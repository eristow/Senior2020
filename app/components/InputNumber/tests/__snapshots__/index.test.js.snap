// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputNumber /> Should render and match the snapshot 1`] = `
<input
  class="InputNumber-sc-1vm18vt-0 dLrMQV"
  type="number"
  width="5em"
/>
`;
